### PR TITLE
Skip tls verify UI

### PR DESCRIPTION
**Pull request by raakasf**
- Added test for backend (not yet complete but working) using testify
- Added tests for frontend (not yet complete but working) using react testing library
- Added backend functionality to skip TLS verification
- Added UI changes to show "skip" button

Without "Skip Verify" the data source fails to fetch health
Without_skip_verify

With "Skip Verify" the data source successfully fetches health
With_skip_verify

Tested with locally updated go-druid library

### CODE REVIEW

File: src/configuration/ConnectionSettings/DruidAuthSettings.tsx [hunk 1]
```diff
@@ -22,8 +29,11 @@ export const DruidAuthSettings: FC<ConnectionSettingsProps> = (props: Connection
         `}
       >
         <Field horizontal label="With basic authentication" description="Enable HTTP Basic authentication">
-          <Switch value={settings.basicAuth} onChange={onSettingChange} />
+          <Switch value={settings.basicAuth} data-testid="basicAuth" name="basicAuth" onChange={onSettingChange} />
```
Comment (jbguerraz) [inline src/configuration/ConnectionSettings/DruidAuthSettings.tsx:32]: we gonna need to add such `data-testid` on elements ? is there any alternative ?

**Review by jbguerraz — CHANGES_REQUESTED** [review]

File: jest.config.js [hunk 2]
```diff
@@ -1,8 +1,6 @@
-// This file is needed because it is used by vscode and other tools that
-// call `jest` directly.  However, unless you are doing anything special
-// do not edit this file
```
[the remark this answers is not in the document]
Comment (raakasf) [reply jest.config.js:3]: I use both VS Code and jest and didn't have any issues. Plus we will need to add something to this file if we want to use jest.

File: src/configuration/ConnectionSettings/DruidAuthSettings.tsx (hunk 1 above)
Comment (raakasf) [reply src/configuration/ConnectionSettings/DruidAuthSettings.tsx:32]: This is one way I found to test, I think as we gain more experience with the react-testing lib we will get better and maybe find a better way too. :)
I have seen other libraries do this too, so probably not the worst thing? (I guess we will know)

File: jest.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbguerraz) [reply jest.config.js:3]: Please, check https://github.com/grafana/grafana/tree/master/packages/grafana-toolkit#how-can-i-test-my-plugin and then https://enzymejs.github.io/enzyme/ 
Looks like the way to go. WDYT ?

File: src/configuration/ConnectionSettings/DruidAuthSettings.tsx (hunk 1 above)
Comment (jbguerraz) [reply src/configuration/ConnectionSettings/DruidAuthSettings.tsx:32]: See https://github.com/grafadruid/druid-grafana/pull/40#discussion_r587296687

File: jest.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (raakasf) [reply jest.config.js:3]: Yes, this looks really good. I've heard of enzyme a lot, but didn't notice that Grafana uses it too. I'll make this change.

**Pull request closed without merging**